src/main/drivers/compass/compass_ak8975.c

```diff
@@ -108,6 +108,7 @@ static bool ak8975Init(magDev_t *mag)
 
     // Trigger first measurement
     busWriteRegister(dev, AK8975_MAG_REG_CNTL, CNTL_BIT_16_BIT | CNTL_MODE_ONCE);
+    mag->magOdrHz = 50; // arbitrary value, need to check what ODR is actually returned
```

Comment:
identical to  ak8963

Reply:
But this mag appears to use a state engine that reports false if not ready, and it looks like after a successful read, it enters `STATE_READ_STATUS1`, and waits.  If we make a read request, while in that state, we will return false, and move to `STATE_READ_STATUS2`. So we can do nothing for 20ms (50hz), attempt a read, if it reports false (most likely `STATE_READ_STATUS1`), we come back 1ms later and most likely return false again (in `STATE_READ_STATUS2`), and then finally we come back 1ms later, the data request is sent, the code in compass.c receives a true flag, and processes the data, returning to `STATE_READ_STATUS1` and waiting about 20ms before we check again.
I'm not sure but that seems correct behaviour to me.

Reply:
Sorry, I though the code is identical after looking on small section only.

The state machine is overkill (and has considerable overhead). It was probably modeled after different Mag.

IMO correct way is to read all data in single read (state1, mag data, state2). If something fails, just ignore data read and start next conversion (this makes sure that Mag is is consistent state. Another option is to read again when conversion did not finish, but handling all edge cases will be quite difficult).
Conversion time is (IIRC) specified as 9ms max, using 100Hz will provide some margin. Mag will be synchronized to readout task (single-shot mode).

Reply:
See comment above.  I'll make a note in the documentation, that support for these is deprecated and that they may not work.